### PR TITLE
mostly style changes, fix links

### PR DESCRIPTION
**Pull request by dirk-thomas**
The patch mostly changes style.
It also updates some outdated links and clarifies the order of some articles.

Also it removes `40_ros_with_modular_middleware`.

@esteve @jacquelinekay  @tfoote @wjwwood For "review".

### CODE REVIEW

**Comment by esteve** [comment]
+1

**Comment by wjwwood** [comment]
Can you rebase since https://github.com/ros2/design/pull/31

**Comment by dirk-thomas** [comment]
Sure: 6a7854ebd5d97151acdb126b10020f5f64a43424

File: articles/20_ros_with_dds.md [hunk 1]
```diff
@@ -236,13 +252,15 @@ Because this conversion is only invoked in conjunction with a more expensive ser
...
+
 ### Services and Actions
 
 DDS currently does not have a ratified or implemented standard for request-response style RPC which could be used to implement the concept of services in ROS.
 There is currently an RPC specification being considered for ratification in the OMG DDS working group, and several of the DDS vendors have a draft implementation of the RPC API.
 It is not clear, however, whether this standard will work for actions, but it could at least support non-preemptable version of ROS services.
 ROS 2.0 could either implement services and actions on top of publish-subscribe (this is more feasible in DDS because of their reliable publish-subscribe QoS setting) or it could use the DDS RPC specification once it is finished for services and then build actions on top, again like it is in ROS 1.x.
-Either way actions will be a first class citizen in the ROS 2.0 API and it may be the case that services just become a degenerate type of action.
+Either way actions will be a first class citizen in the ROS 2.0 API and it may be the case that services just become a degenerated type of actions.
+
```
Comment (wjwwood) [inline articles/20_ros_with_dds.md:263]: `degenerate` is the right tense here. Maybe a clearer statement would be `services just become a degenerate form of actions.`

File: articles/10_why_ros2.md [hunk 2]
```diff
@@ -3,7 +3,8 @@ layout: default
 title: Why ROS 2.0?
 permalink: articles/why_ros2.html
 abstract:
-  This article captures the reasons for making breaking changes to the ROS API, hence the 2.0.
+  This article captures the reasons for making breaking changes to the ROS API,
+  hence the 2.0.
```
Comment (wjwwood) [inline articles/10_why_ros2.md]: Why do this? It seems to be the reverse operation as the other changes in this file.

**Comment by wjwwood** [comment]
+1 other than comments.

File: articles/20_ros_with_dds.md (hunk 1 above)
Comment (jackieokay) [inline articles/20_ros_with_dds.md:263]: If you're a mathematician, the correct wording is: "services become a degenerate _case_ of actions"

http://en.wikipedia.org/wiki/Degeneracy_(mathematics)

File: articles/10_why_ros2.md (hunk 2 above)
Comment (dirk-thomas) [inline articles/10_why_ros2.md]: For no specific reason. Reverted this change.

File: articles/20_ros_with_dds.md (hunk 1 above)
Comment (dirk-thomas) [inline articles/20_ros_with_dds.md:263]: Sounds good to me. Done.

**Comment by dirk-thomas** [comment]
If there are more comments later I can address them after the merge.

**Pull request merged**